zerver/lib/actions.py

```diff
@@ -5671,6 +5672,49 @@ def do_update_message_flags(
         "messages": messages,
         "all": False,
     }
+
+    if flag == "read" and operation == "remove":
+        raw_unread_data = get_raw_unread_data(user_profile, messages)
```

Comment:
This is a substantial enough chunk of code that we should probably put it into its own function in `lib/messages.py`.